From shemminger@osdl.org Fri Oct 06 17:10:48 2006
Received: (at 391382) by bugs.debian.org; 7 Oct 2006 00:10:48 +0000
Return-path: <shemminger@osdl.org>
Received: from smtp.osdl.org ([65.172.181.4])
	by spohr.debian.org with esmtp (Exim 4.50)
	id 1GVzmC-0008SO-GB
	for 391382@bugs.debian.org; Fri, 06 Oct 2006 17:10:48 -0700
Received: from shell0.pdx.osdl.net (fw.osdl.org [65.172.181.6])
	by smtp.osdl.org (8.12.8/8.12.8) with ESMTP id k970AiaX032159
	(version=TLSv1/SSLv3 cipher=EDH-RSA-DES-CBC3-SHA bits=168 verify=NO);
	Fri, 6 Oct 2006 17:10:45 -0700
Received: from freekitty (freekitty.pdx.osdl.net [10.8.0.54])
	by shell0.pdx.osdl.net (8.13.1/8.11.6) with ESMTP id k970Ae2c018518;
	Fri, 6 Oct 2006 17:10:40 -0700
Date: Fri, 6 Oct 2006 16:09:12 -0700
From: Stephen Hemminger <shemminger@osdl.org>
To: Oleg Verych <olecom@flower.upol.cz>
Cc: Matthias Kreis <Matthias.Kreis@remasec.ch>, 391382@bugs.debian.org
Subject: Re: sky2 stops working 2.6.17 or ethernet driver crashes 2.6.18
Message-ID: <20061006160912.1d576012@freekitty>
In-Reply-To: <slrneicn33.43t.olecom@deen.upol.cz.local>
References: <20061006110110.10879.23251.reportbug@rincewind.remasec.local>
	<slrneicn33.43t.olecom@deen.upol.cz.local>
Organization: OSDL
X-Mailer: Sylpheed-Claws 2.5.0-rc3 (GTK+ 2.10.6; i486-pc-linux-gnu)
Mime-Version: 1.0
Content-Type: text/plain; charset=US-ASCII
Content-Transfer-Encoding: 7bit
X-MIMEDefang-Filter: osdl$Revision: 1.155 $
X-Scanned-By: MIMEDefang 2.36
X-Spam-Checker-Version: SpamAssassin 2.60-bugs.debian.org_2005_01_02 
	(1.212-2003-09-23-exp) on spohr.debian.org
X-Spam-Level: 
X-Spam-Status: No, hits=-1.4 required=4.0 tests=BAYES_00,UPPERCASE_25_50 
	autolearn=no version=2.60-bugs.debian.org_2005_01_02

Does this help.. it is going in next patchset.
--------------------------------------------------------

Subject: sky2: fix hangs on some chips

The driver inherited some bad setup code from sk98lin (vendor) driver that
causes receive FIFO probelms.  Just remove it.

Signed-off-by: Stephen Hemminger <shemminger@osdl.org>

--- sky2.orig/drivers/net/sky2.c	2006-10-06 15:53:52.000000000 -0700
+++ sky2/drivers/net/sky2.c	2006-10-06 15:57:47.000000000 -0700
@@ -689,15 +689,10 @@
 	sky2_write8(hw, SK_REG(port, TX_GMF_CTRL_T), GMF_RST_CLR);
 	sky2_write16(hw, SK_REG(port, TX_GMF_CTRL_T), GMF_OPER_ON);
 
-	if (hw->chip_id == CHIP_ID_YUKON_EC_U) {
-		sky2_write8(hw, SK_REG(port, RX_GMF_LP_THR), 768/8);
-		sky2_write8(hw, SK_REG(port, RX_GMF_UP_THR), 1024/8);
-		if (hw->dev[port]->mtu > ETH_DATA_LEN) {
-			/* set Tx GMAC FIFO Almost Empty Threshold */
-			sky2_write32(hw, SK_REG(port, TX_GMF_AE_THR), 0x180);
-			/* Disable Store & Forward mode for TX */
-			sky2_write32(hw, SK_REG(port, TX_GMF_CTRL_T), TX_STFW_DIS);
-		}
+	if (hw->chip_id == CHIP_ID_YUKON_EC_U
+	    && hw->dev[port]->mtu > ETH_DATA_LEN) {
+		/* Disable Store & Forward mode for TX */
+		sky2_write32(hw, SK_REG(port, TX_GMF_CTRL_T), TX_STFW_DIS);
 	}
 
 }



